collector/thermalzone.go

```diff
@@ -0,0 +1,105 @@
+package collector
+
+import (
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+func init() {
+	Factories["thermalzone"] = NewThermalZoneCollector
+}
+
+// A ThermalZoneCollector is a Prometheus collector for WMI Win32_PerfRawData_Counters_ThermalZoneInformation metrics
+type ThermalZoneCollector struct {
+	HighPrecisionTemperature *prometheus.Desc
+	PercentPassiveLimit      *prometheus.Desc
+	Temperature              *prometheus.Desc
+	ThrottleReasons          *prometheus.Desc
+}
+
+// NewThermalZoneCollector ...
+func NewThermalZoneCollector() (Collector, error) {
+	const subsystem = "thermalzone"
+	return &ThermalZoneCollector{
+		HighPrecisionTemperature: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "high_precision_temperature"),
+			"(HighPrecisionTemperature)",
+			nil,
+			nil,
+		),
+		PercentPassiveLimit: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "percent_passive_limit"),
+			"(PercentPassiveLimit)",
+			nil,
+			nil,
+		),
+		Temperature: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "temperature"),
```

Comment:
As with the high-precision one, suffix with unit. Do we really need both?

Reply:
I think it would be good to keep high precision, but call it Temperature. Not sure why one would want less precision in any case. From what I can tell it's only 1 decimal

Reply:
Agree. I suspect it is the usual backwards compatibility stuff - at some point they could get better precision, but then there's a bunch of code out there assuming it it an integer value in whole degrees, etc.